aurora-locker/src/Locker.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: CC-BY-1.0
+pragma solidity ^0.8.17;
+
+import "openzeppelin-contracts/token/ERC20/IERC20.sol";
+import "openzeppelin-contracts/token/ERC20/extensions/IERC20Metadata.sol";
+import "./AuroraSdk.sol";
+
+string constant ERR_METHOD_NOT_IMPLEMENTED = "ERR_METHOD_NOT_IMPLEMENTED";
+// TODO: Determine proper values for gas.
+uint64 constant ON_DEPOSIT_NEAR_GAS = 3_000_000_000_000;
+// TODO: Determine proper values for gas.
+uint64 constant DEPOSIT_CALLBACK_NEAR_GAS = 3_000_000_000_000;
+
+// TODO: Implement Pause mechanics.
+// TODO: Implement Upgradable mechanics.
+// TODO: Implement AdminControlled mechanics.
+
+/// The Locker contract holds all ERC20 tokens that are sent from Aurora to NEAR.
+/// It provides an interface for depositing and withdrawing the tokens. After deposit
+/// the same amount of tokens are minted on the equivalent NEP141 contract. In a similar
+/// way, when tokens are burnt on NEAR side, they are withdrawn back to the specified
+///  user on Aurora.
+contract Locker {
+    using AuroraSdk for NEAR;
+    using AuroraSdk for PromiseWithCallback;
+    using AuroraSdk for PromiseCreateArgs;
+    using Codec for bytes;
+    using Codec for uint128;
+
+    /// NEAR Account Id of the factory
+    string public factoryAccountId;
+    /// Implicit addres for the factory
+    address public factoryImplicitAddress;
+    /// Interface to interact with NEAR smart contracts.
+    NEAR public near;
+    /// Implicit address for representative NEAR account of this contract.
+    address public immutable selfReprsentativeImplicitAddress;
+
+    constructor(string memory factoryAccountId_) {
+        factoryAccountId = factoryAccountId_;
+        factoryImplicitAddress = AuroraSdk.implicitAuroraAddress(factoryAccountId);
+        near = AuroraSdk.mainnet();
```

Comment:
I wonder if there should be an alternate constructor which allows specifying the NEAR parameters explicitly as well. It would be useful in tests.

Reply:
Done in https://github.com/aurora-is-near/native-erc20-connector/pull/8/commits/6499754bfd8571f5fd0183302e3b4c778e688da5